troposphere/static/js/components/modals/tag/TagCreateModal.jsx

```diff
@@ -3,132 +3,53 @@ import $ from "jquery";
 import stores from "stores";
 import BootstrapModalMixin from "components/mixins/BootstrapModalMixin";
 
-// Example Usage from http://bl.ocks.org/insin/raw/8449696/
-// render: function(){
-// <div>
-//   ...custom components...
-//   <ExampleModal
-//      ref="modal"
-//      show={false}
-//      header="Example Modal"
-//      buttons={buttons}
-//      handleShow={this.handleLog.bind(this, 'Modal about to show', 'info')}
-//      handleShown={this.handleLog.bind(this, 'Modal showing', 'success')}
-//      handleHide={this.handleLog.bind(this, 'Modal about to hide', 'warning')}
-//      handleHidden={this.handleLog.bind(this, 'Modal hidden', 'danger')}
-//    >
-//      <p>I'm the content.</p>
-//      <p>That's about it, really.</p>
-//    </ExampleModal>
-// </div>
-//
-
-// To show the modal, call this.refs.modal.show() from the parent component:
-// handleShowModal: function() {
-//   this.refs.modal.show();
-// }
-
-function getState() {
-    return {
-        name: null,
-        description: null
-    }
-}
-
 export default React.createClass({
     displayName: "TagCreateModal",
 
     mixins: [BootstrapModalMixin],
 
     propTypes: {
-        initialTagName: React.PropTypes.string
+        name: React.PropTypes.string
     },
 
-    isSubmittable: function() {
-        var hasName = !!this.state.name;
-        var hasDescription = !!this.state.description;
-        var tagExists = !!this.state.tagExists;
-        return hasName && hasDescription && !tagExists;
+    getInitialState: function() {
+        // state.name will only be props.name initially, this is intentional
+        // because we are using to it to set the initial default
+        return {
+            name: this.props.name || "",
+            description: ""
+        };
     },
 
     componentDidMount: function() {
-        if (this.state.name) {
-            let lower = this.state.name.toLowerCase();
-            let tags = stores.TagStore.getAll().filter(function(tag) {
-                return tag.get("name").toLowerCase() === lower;
-            });
-            if (tags.length > 0) {
-                this.setState({
-                    tagExists: true,
-                    existsText: "Tag " + this.state.name + " already exists"
-                });
-            } else {
-                this.setState({
-                    tagExists: false,
-                    existsText: ""
-                });
-            }
-        }
+        stores.TagStore.addChangeListener(this.updateState);
     },
 
-    //
-    // Mounting & State
-    // ----------------
-    //
-
-    getInitialState: function() {
-        return {
-            name: this.props.initialTagName,
-            description: null
-        }
+    componentWillUnmount() {
+        stores.TagStore.removeChangeListener(this.updateState);
     },
 
     updateState: function() {
-        if (this.isMounted()) this.setState(getState());
+        if (this.isMounted()) {
```

Comment:
We have to use `isMounted`. Because modals are closed with `hide` and then they're `onConfirm` method is called after. The `onConfirm` will typically set some state on the component that was just closed and trigger a warning.